packages/matchbox/src/components/Table/Table.js

```diff
@@ -1,39 +1,60 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { Cell, HeaderCell, Row } from './TableElements';
-import styles from './Table.module.scss';
-
-class Table extends Component {
-  static displayName = 'Table';
-
-  static Cell = Cell;
-  static HeaderCell = HeaderCell;
-  static Row = Row;
-
-  static propTypes = {
-    data: PropTypes.array,
-    /**
-     * React node(s)
-     */
-    children: PropTypes.node
-  };
-
-  render() {
-    const {
-      children,
-      data
-    } = this.props;
-
-    const dataMarkup = data
-      ? <tbody>{data.map((rowData, i) => <Row rowData={rowData} key={`Row-${i}`} />)}</tbody>
-      : children;
-
-    return (
-      <table className={styles.Table}>
+import styled from 'styled-components';
+import { margin, padding, compose } from 'styled-system';
+import { createPropTypes } from '@styled-system/prop-types';
+import { table } from './styles';
+import { pick } from '@styled-system/props';
+
+const system = compose(margin, padding);
+
+const StyledTable = styled('table')`
+  ${table}
+  ${system}
+`;
+
+export const TablePaddingContext = React.createContext({});
```

Comment:
I think this might be a circular dependency – the component `import { TablePaddingContext } from './Table';` but then this file also has `import { Cell, HeaderCell, Row } from './TableElements';`

might be better to split this up into a new file